draft-ietf-quic-multipath.md

```diff
@@ -885,6 +884,37 @@ two or more active paths during the connection lifetime. Different applications
 Once the implementation has decided which paths to keep alive, it can do so by sending Ping frames
 on each of these paths before the idle timeout expires.
 
+## Connection ID Changes and NAT Rebindings
+
+{{Section 5.1.2 of QUIC-TRANSPORT}} indicates that an endpoint
+can change the Connection ID it uses for to another available one
+at any time during the connection. As such a sole change of the Connection
+ID without any change in the address does not indicate a path change and
+the endpoint will keep the same congestion control and RTT measurement state.
+
+While endpoints assign a Connection ID to a specific sending 4-tuple,
+networks events such as NAT rebinding may make the packet's receiver
+observe a different 4-tuple. Servers observing a 4-tuple change will
+performs path validation (see {{Section 9 of QUIC-TRANSPORT}}).
+If the path validation process succeeds, the endpoints need to reset
+their congestion controller and round-trip time estimator for the new
+path as specficied in  {{Section 9.4 of QUIC-TRANSPORT}}.
+
+{{Section 9.3 of QUIC-TRANSPORT}} allows an endpoint to skip validation of
+a peer address if that address has been seen recently. However, when the
+multipath extension is used and an endpoint has multiple addresses that
+could lead to switching between different paths, it should rather maintain
+multiple open paths instead.
+
+If an endpoint uses a new Connection ID after an idle period
+and a NAT rebinding leads to a 4-tuple changes on the same packet,
+the receiving endpoint may not be able to associate the packet to
+an existing path and will therefore consider this as a new path.
+This leads to an inconsistent view of open paths at both peers,
+however, as the "old" path will not work anymore, it will be silently
+closed after the idle timeout expires.
+
```

Comment:
I think overloading the meaning of PATH_CHALLENGE is a bad idea, see comments on #188. But sending a "RETIRE_CONNECTION_ID" too soon is also a bad idea, because the peer needs some time to send acknowledgements. Maybe the peer doing the CID renewal could send PATH_ABANDON(old_path, reason_code = CID_replaced), with a clear statement explaining that the 4-tuple is still valid, but the CID is not?

Reply:
Let's open an issue to add a reason code, or a replaced by field, or both. Maybe we should also rename path_abandon to cid_abandon?

Question is should we merge this text for now or also open a new issue instead? I think we can merge because it shouldn't hurt to send a path_abandon frame, right?

Reply:
I think this issue is somewhat orthogonal, and discussed in #188.

Reply:
We do have an issue with the server continuing sending data on the old path, now a dead end. I am concerned that we will not have an agreement before Monday. I left comments in issue #188. If we resolve that issue before Friday 3/10, let's incorporate the agreed solution in this PR. If we do not, let's check in the PR as is, without the extra text.